core/src/main/scala/cats/Selective.scala

```diff
@@ -27,12 +28,46 @@ trait Selective[F[_]] {
     branch(condition)(left)(right)
   }
 
-  // TODO more combinators here
+  def whenS[A](fbool: F[Boolean])(fa: F[Unit]): F[Unit] =
+    ifS(fbool)(Functor[F].as(fa, ()))(().pure[F])
+
```

Comment:
By the way @snowleopard defining `whenS` in terms of `ifS` like this seems slightly simpler than defining it in terms of `select` like you do in the paper. Was that just because you wanted to introduce `whenS` earlier in the paper than `ifS`?

Reply:
Yes, exactly: `whenS` is a very simple conditional combinator, which is almost `select` in spirit, so it felt better to avoid introducing `ifS` in between.